seasmon_xr/accessors.py

```diff
@@ -78,6 +113,60 @@ def _gen_labels(x, lbl, c):
         return f"{x.year}{x.month:02}" + f"{lbl}{int(x.day//c+1)}"
 
 
+class Period(AccessorTimeBase):
+    # pylint: disable=no-member,undefined-variable
+    """
+    Baseclass to extend time dimension with period functionality.
+
+    Adds functionality for working with periods, such as dekads and pentads
+    """
+
+    @property
+    def month_idx(self):
```

Comment:
In a first try, I used `{month,year}_ix` as property name as it felt bit more natural than `n_{year,month}` and you can more easily access first e.g. `year` and followed by `year_ix` in an interactive setting. But happy to change to a better naming.

Reply:
I think both names are kinda confusing, `month_idx` in this case means "dekad index within a month, 1,2 or 3", but can also be interpreted as "index of the month, 1-12". Same problem with `n_month`.

Reply:
Also I'm not too sure about the use of those magic constants (`_cfactor`), I think this can be defined in a more obvious way with just two integer parameters:
- number of days per period (5, 10), let's say `_ndays`
- maximum allowed periods per month (6, 3), (which is really `int(31/_ndays)`)

Then index within a month becomes:
```python
min(_max_per_month, (day // _ndays) + 1))
```

Reply:
Back no naming things. I think what we really have here is an "index of a given period within a year". But it has several representations:

- **flat index view**:  1 or 0 based index, for dekad this would be: `[1, 36]` or `[0, 35]`
- **structured view**: `(month, month_idx)`

I think `month_idx` and `year_idx` would be better off named `midx, yidx`. I think this is a situation where using "more human" names leads to ambiguity, one is more likely to assume they understand what `month_idx` is, even if they picked the wrong interpretation out of the two possible interpretations.

Reply:
> Also I'm not too sure about the use of those magic constants (`_cfactor`), I think this can be defined in a more obvious way with just two integer parameters:
> 
> * number of days per period (5, 10), let's say `_ndays`
> * maximum allowed periods per month (6, 3), (which is really `int(31/_ndays)`)
> 
> Then index within a month becomes:
> 
> ```python
> min(_max_per_month, (day // _ndays) + 1))
> ```


That's fine, at one point I had a similar approach but didn't like the fact we need to add `min` too much ... but happy to change to this

Reply:
there is an error there, should be

```python
min(_max_per_month, ((day-1) // _ndays) + 1))
```

10-th of the month is the first dekad, right?

Reply:
Yes, 1-10 = d1, 11-20 = d2, 21-EOM = d3